analyzer-models/src/main/java/ffb/analyzer/models/espn/TeamRecordBase.java

```diff
@@ -0,0 +1,60 @@
+package ffb.analyzer.models.espn;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public abstract class TeamRecordBase extends EspnEntity<TeamRecordBase> {
```

Comment:
Does this need to be `public`?

I also think we'll want this to be `TeamRecordBase<T extends TeamRecordBase> extends EspnEntity<T>`.
Doesn't matter right now since `EspnEntity` doesn't provide any functionality, but my thought is that `EspnEntity` has knowledge of the concrete implementation.

Reply:
Both are good points. Since it's an abstract class meant to pass down reusable info to the other TeamRecord* types, there's no need for it to be public.

Also makes sense to separate the concrete implementation from EspnEntity. I've updated the TeamRecordBase class declaration, and each of the inheriting TeamRecord* classes extend TeamRecordBase with their appropriate type. From reading up on Java Generics, I believe we're looking for Recursive Generics, right? Which would require an additional <T> in the declaration.